### PR TITLE
[MNT] Test pandas<2.1.0

### PR DESCRIPTION
**Pull request by aiwalter**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ dependencies = [
     "deprecated>=1.2.13",
     "numba>=0.55",
     "numpy>=1.21.0,<1.25",
-    "pandas>=1.1.0,<1.6.0",
+    "pandas==2.0.0rc0",
```
Comment (aiwalter) [inline pyproject.toml:44]:
```suggestion
    "pandas==2.0.0rc1",
```

**Comment by aiwalter** [comment]
see https://github.com/predict-idlab/plotly-resampler/issues/193

File: aeon/transformations/panel/tests/test_interpolate.py [hunk 2]
```diff
@@ -23,7 +25,7 @@ def test_resizing():
     3) use transformer for resizing to resize time series to equal length
     4) result lengths are equal to length that was set for transformer
     """
-    X, _ = load_basic_motions(split="train", return_type="nested_univ")
+    X, _ = load_basic_motions(split="train", return_X_y=True, return_type="nested_univ")
```
Comment (GuzalBulatova) [inline aeon/transformations/panel/tests/test_interpolate.py:28]: Not relevant for this PR, but what is "nested_univ"? Does it essentially mean that it consists of just one col? Is there a reason to specify that the nested DF is univariate? 🤔

File: aeon/datasets/_data_io.py [hunk 3]
```diff
@@ -2035,11 +2035,10 @@ def _convert_tsf_to_hierarchical(
...
     index_columns = [c for c in list(df.columns) if c not in drop_columns + columns]
     result = pd.DataFrame({c: df[c].explode() for c in columns})
-    df = (
-        df.drop(columns=columns + drop_columns)
-        .join(result)
-        .set_index(index_columns + ["timestamp"])
-    )
+    df = df.drop(columns=columns + drop_columns).join(result)
+    if df["timestamp"].dtype == "object":
+        df = df.astype({"timestamp": "int64"})
```
Comment (GuzalBulatova) [inline aeon/datasets/_data_io.py:2040]: Is object here assumed to always be convertible to integer? Maybe worth including a value error, what do you think?

**Review by GuzalBulatova — APPROVED** [review]
Looks great to me, amazing work, Martin! None of my comments is a blocker to merge this.

File: aeon/transformations/panel/tests/test_interpolate.py (hunk 2 above)
Comment (TonyBagnall) [inline aeon/transformations/panel/tests/test_interpolate.py:28]: the default for load_basic_motions return_X_y is true, you dont need to add this

**Pull request merged**